docs/guide/editing-custom-widgets-in-context.md

```diff
@@ -0,0 +1,288 @@
+
+### Debouncing update events
+
+If the value will be changing quickly, for instance as the user types, performance can suffer if an `update` event is generated for every keystroke. The widget editor can speed up the interface by emitting an `update` event no more than once per second. This is called "debouncing."
+
+As a hint that input might not yet be saved, a widget that "debounces" `update` events should emit a `context-editing` event on every change, even if `update` is intentionally delayed. However this should only be done if the `docId` prop passed to the widget editor is equal to `windows.apos.adminBar.contextId`.
+
+If the `focused` prop of the widget editor becomes `false` and an `update` event has been delayed for debouncing purposes, the widget should cancel its timer and immediately emit the update. Use Vue's `watch` feature to monitor `focused`.
+
+### Saving the content
+
+By default the `fields` section of a contextually edited widget has no effect on the user interface. Emitting a reasonable value is the task of the custom widget editor component. However configuring fields is still necessary since the server will still use them to sanitize the data before saving it.
+
+If no `fields` section is configured, no data will be saved at all unless the `sanitize` method of the module is overridden to provide an alternative way to verify user input, as is done in the case of the core rich text widget.
+
+## Example
+
+Here is a simple "hero" heading widget. Both the heading and the width of the heading can be adjusted directly on the page. This gives the user immediate visual feedback.
+
+<AposCodeBlock>
+```js
+module.exports = {
+  extend: '@apostrophecms/widget-type',
+  options: {
+    contextual: true,
+    defaultData: {
+      heading: '',
+      width: 100
+    },
+    components: {
+      widgetEditor: 'HeroWidgetEditor'
+    }
+  },
+  fields: {
+    add: {
+      heading: {
+        type: 'string',
+        def: ''
+      },
+      width: {
+        type: 'integer',
+        min: 10,
+        max: 100,
+        def: 100
+      }
+    }
+  }
+};
+```
+  <template v-slot:caption>
+    modules/hero-widget/index.js
+  </template>
+</AposCodeBlock>
+
+**What is happening here?**
+
+This `index.js` file configures the `hero-widget` module. The `contextual: true` option enables contextual editing, `defaultData` provides a default value that satisfies the schema, and `components` configures the Vue component name to be used for editing.
+
+The `fields` section provides a schema of fields to sanitize and store the data in the widget. Even though the interface does not use a traditional form, this is still a convenient way to verify the content meets expectations on the server side. Remember: it is never OK for the server to trust the browser.
+
```

Comment:
This would be better visualized using a table, like
<img width="853" alt="image" src="https://user-images.githubusercontent.com/1889830/161985104-0a268ce6-2bcd-440d-babb-45f1bfa4791e.png">

Reply:
`fields` is a section while the others are options (nested in `options`), so I'm concerned that a table would imply they are all on one level.